wavefront/server/packages/flo_cloud/flo_cloud/azure/storage_queue.py

```
@@ -0,0 +1,107 @@
+    Event Grid delivers messages to Storage Queue as base64-encoded JSON.
+    Messages we send ourselves are plain JSON.
+    """
+    try:
+        return json.loads(content)
+    except json.JSONDecodeError:
+        try:
+            return json.loads(base64.b64decode(content).decode('utf-8'))
+        except (binascii.Error, UnicodeDecodeError) as e:
+            raise ValueError(
+                f'Message content is neither valid JSON nor base64-encoded JSON: {e}'
+            )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)